src/NuGet.Core/NuGet.Packaging/PackageExtraction/PackageHelper.cs

```diff
@@ -68,12 +66,18 @@ public static bool IsPackageFile(string packageFileName, PackageSaveMode package
             {
                 return !ExcludePaths.Any(p =>
                     packageFileName.StartsWith(p, StringComparison.OrdinalIgnoreCase)) &&
-                    !packageFileName.EndsWith(ExcludeExtension, StringComparison.OrdinalIgnoreCase);
+                    !IsHashFile(packageFileName);
             }
 
             return false;
         }
 
+        private static bool IsHashFile(string path)
```

Comment:
This name seems misleading. The new file is not exclusively a hash file, but a metadata file that happens to have a hash on it.